assert/assertions.go

```diff
@@ -1333,8 +1335,8 @@ func diff(expected interface{}, actual interface{}) string {
 		return ""
 	}
 
-	et, ek := typeAndKind(expected)
-	at, _ := typeAndKind(actual)
+	et, ek, expectedIsPointer := typeAndKind(expected)
+	at, _, actualIsPointer := typeAndKind(actual)
```

Comment:
@pwfcurry Instead of adding `bool` and changing `typeAndKind` func, we could check `ek` and `ak` is reflect.Ptr.

Could you add the testcase to verify the output, when we deal with pointers.

Reply:
ek/ak are strings at this point, as we're returning `reflect.TypeOf(*str).Kind()`.

I've added tests in assert/assertions_test.go